include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce.h

```diff
@@ -0,0 +1,387 @@
+                    [=](sycl::nd_item<1> __item_id) {
+                        ::std::size_t __global_idx = __item_id.get_global_id(0);
+                        ::std::uint16_t __local_idx = __item_id.get_local_id(0);
+                        // 1. Initialization (transform part). Fill local memory
+                        if (__is_first)
+                        {
+                            __transform_op1.parallel(__local_idx, __n, __iters_per_work_item, __global_idx,
+                                                     /*global_offset*/ 0, __temp_local, __rngs...);
+                        }
+                        else
+                        {
+                            __transform_op2.parallel(__local_idx, __n, __iters_per_work_item, __global_idx, __offset_2,
+                                                     __temp_local, __temp_acc);
+                        }
+                        __dpl_sycl::__group_barrier(__item_id);
+                        // 2. Reduce within work group using local memory
+                        _Tp __result = __reduce_op(__item_id, __global_idx, __n_items, __temp_local);
+                        if (__local_idx == 0)
+                        {
+                            // final reduction
+                            if (__n_groups == 1)
+                            {
+                                __reduce_op.apply_init(__init, __result);
+                                __res_acc[0] = __result;
+                            }
+
+                            __temp_acc[__offset_1 + __item_id.get_group(0)] = __result;
+                        }
+                    });
+            });
+            if (__is_first)
+                __is_first = false;
+            ::std::swap(__offset_1, __offset_2);
+            __n = __n_groups;
+            __n_items = __ceiling_div(__n, __iters_per_work_item);
+            __n_groups = __ceiling_div(__n, __size_per_work_group);
+        } while (__n > 1);
+
+        return __future(__reduce_event, __res);
+    }
+}; // struct __parallel_transform_reduce_submitter
+
+// General version of parallel_transform_reduce. Calls optimized kernels.
+template <typename _Tp, typename _ReduceOp, typename _TransformOp, typename _ExecutionPolicy, typename _InitType,
+          oneapi::dpl::__internal::__enable_if_device_execution_policy<_ExecutionPolicy, int> = 0, typename... _Ranges>
+auto
+__parallel_transform_reduce(_ExecutionPolicy&& __exec, _ReduceOp __reduce_op, _TransformOp __transform_op,
+                            _InitType __init, _Ranges&&... __rngs)
+{
+    auto __n = oneapi::dpl::__ranges::__get_first_range_size(__rngs...);
+    assert(__n > 0);
+
+    if (__n <= 64)
+        return __parallel_transform_reduce_seq_submitter<_Tp, _ReduceOp, _TransformOp>(
+            ::std::forward<_ExecutionPolicy>(__exec), __n, __reduce_op, __transform_op, __init,
+            ::std::forward<_Ranges>(__rngs)...);
+    else
```

Comment:
Here "else" is not necessary because we have `return`  in `if ` scope. 
An advantage of  removing "else"  - we have not {} and indents for the whole "else" code block.
But it is up to you.

Reply:
I think this makes the code easier to read and I will change it.